Makefile

```diff
@@ -26,24 +26,29 @@ test: test-unit test-integration test-e2e ## Run all tests
 test-unit: ## Run the unit tests
 	@echo "Running unit tests..."
 	@echo "Reports will be written to ${ARTIFACT_DIR}"
-	python -m pytest tests/unit --cov=ols --cov-report term-missing --cov-report "json:${ARTIFACT_DIR}/coverage_unit.json" --junit-xml="${ARTIFACT_DIR}/junit_unit.xml"
+	COVERAGE_FILE=.coverage.unit python -m pytest tests/unit --cov=ols --cov-report term-missing --cov-report "json:${ARTIFACT_DIR}/coverage_unit.json" --junit-xml="${ARTIFACT_DIR}/junit_unit.xml"
 	python scripts/transform_coverage_report.py "${ARTIFACT_DIR}/coverage_unit.json" "${ARTIFACT_DIR}/coverage_unit.out"
 	scripts/codecov.sh "${ARTIFACT_DIR}/coverage_unit.out"
 
 test-integration: ## Run integration tests tests
 	@echo "Running integration tests..."
 	@echo "Reports will be written to ${ARTIFACT_DIR}"
-	python -m pytest tests/integration --cov=ols --cov-report term-missing --cov-report "json:${ARTIFACT_DIR}/coverage_integration.json" --junit-xml="${ARTIFACT_DIR}/junit_integration.xml" --cov-fail-under=60
+	COVERAGE_FILE=.coverage.integration python -m pytest tests/integration --cov=ols --cov-report term-missing --cov-report "json:${ARTIFACT_DIR}/coverage_integration.json" --junit-xml="${ARTIFACT_DIR}/junit_integration.xml" --cov-fail-under=60
 	python scripts/transform_coverage_report.py "${ARTIFACT_DIR}/coverage_integration.json" "${ARTIFACT_DIR}/coverage_integration.out"
 	scripts/codecov.sh "${ARTIFACT_DIR}/coverage_integration.out"
 
+check-coverage: test-unit test-integration  ## Unit tests and integration tests overall code coverage check
+	coverage combine --keep
+	# the threshold should be very high there, in theory it should reach 100%
+	coverage report -m --fail-under=94
+
 test-e2e: ## Run e2e tests - requires running OLS server
 	@echo "Running e2e tests..."
 	@echo "Reports will be written to ${ARTIFACT_DIR}"
 	python -m pytest tests/e2e --junit-xml="${ARTIFACT_DIR}/junit_e2e.xml"
 		
 
-coverage-report:	test-unit ## Export unit test coverage report into interactive HTML
+coverage-report:	.coverage_unit  ## Export unit test coverage report into interactive HTML
```

Comment:
can you explain this change?  there is no ".coverage_unit" target, nor does any step seem to create that file.

Reply:
`COVERAGE_FILE=.coverage.unit python -m pytest ...` command creates it, it's `test-unit` target

Reply:
hmm thinking about it more, I'll just use the original line 46 as is. It was an optimization not to run tests again when the coverage file exists. But yes, as you said - Make does not know if the file is refreshed or not (not having knowledge that every single line in the project can change it basically).

Reply:
updated @bparees